test/build/runner.js

```diff
@@ -0,0 +1,79 @@
+const assert = require('assert');
+const path = require('path');
+const fs = require('fs').promises;
+const {readdirSync, readFileSync, statSync, existsSync} = require('fs');
+const execa = require('execa');
+const dircompare = require('dir-compare');
+
+function stripWhitespace(stdout) {
+  return stdout.replace(/((\s+$)|((\\r\\n)|(\\n)))/gm, '');
+}
+function stripRev(code) {
+  return code.replace(/\?rev=\w+/gm, '?rev=XXXXXXXXXX');
+}
+function stripChunkHash(stdout) {
+  return stdout.replace(/([\w\-]+\-)[a-z0-9]{8}(\.js)/g, '$1XXXXXXXX$2');
+}
+
+for (const testName of readdirSync(__dirname)) {
+  if (testName === 'node_modules' || testName.includes('.')) {
+    continue;
+  }
+
+  test(testName, async () => {
+    let cwd = path.join(__dirname, testName);
+
+    // add exception for yarn-workspaces test
+    if (testName === 'yarn-workspaces') {
+      cwd = path.join(cwd, 'packages', 'snowpack');
+    }
+
+    // Run Test
+    const {all} = await execa('npm', ['run', `TEST`, `--silent`], {
+      cwd,
+      reject: false,
```

Comment:
We do `reject: false` in the install test runner because we want to compare the failed output to the expected output.

If there's no similar test here, then have reject true so that failed builds fail the execa test